Subway/Subway/Loader.cs

```diff
@@ -12,7 +12,62 @@ public class Loader
         public Dictionary<string, Line> Load(string filePath)
         {
             var lines = new Dictionary<string, Line>();
+
+            string record;
+            string lineName = "";
+            using (StreamReader fileReader = new StreamReader(filePath))
+            {
+                if ((record = fileReader.ReadLine()) != null)
```

Comment:
Please review the logic inside this method. You're using the same condition three times in 'if' and 'while' statements.